### ISSUE TITLE
Improve support for `removeparam` in standard mode

### ISSUE THREAD

**Issue by fmarier** [brave/brave-browser#35141]
Besides being used as privacy-harming tracking vectors (covered by [Brave's own filter](https://github.com/brave/brave-browser/wiki/Query-String-Filter) since 2019), query string parameters are also removed for other reasons in custom rule sets, and are involved in certain kinds of ads.

At the moment, support for `removeparam` rules in adblock lists is gated on users being in aggressive mode which means that:
- `removeparam` rules added by users (directly in `brave://settings/shields/filters` or via a custom list) will be silently ignored unless the user also opts into aggressive mode
- Video ads from SSAI (server-side ad insertion) systems will not be blocked in standard mode

Perhaps we can fix both of these limitations by adding a new metadata field to https://github.com/brave/adblock-resources/blob/master/filter_lists/list_catalog.json and gating `removeparam` on that instead.

**Comment by fmarier** [comment]
For the release notes: Add support for `removeparam` in custom rules when adblocking is set to standard mode.

**Comment by stephendonner** [comment]
## Verified `PASSED` using

```
Brave | 1.63.121 Chromium: 121.0.6167.57 (Official Build) nightly (x86_64)
-- | --
Revision | 875b8d7b36b12cd0707de7d638e05f9da6238ef5
OS | macOS Version 11.7.10 (Build 20G1427)
```

### Steps:
1. installed `1.63.121`
2. launched Brave
3. loaded `brave://adblock`
4. added `||example.com^$removeparam=test` to the `Custom Filters` section
5. clicked `Save`
6. loaded `https://example.com/?test=true`

### Confirmed that the `?test=true` param was removed

example | example | example
---------|----------|---------
<img width="1577" alt="Screen Shot 2024-01-16 at 9 35 16 AM" src="https://github.com/brave/brave-browser/assets/387249/1f089108-c707-4e21-800b-8c15c7dd7b9e"> | <img width="1944" alt="Screen Shot 2024-01-16 at 9 36 25 AM" src="https://github.com/brave/brave-browser/assets/387249/f98bfc47-c142-4647-8372-5a941e210bfc"> | <img width="1944" alt="Screen Shot 2024-01-16 at 9 36 48 AM" src="https://github.com/brave/brave-browser/assets/387249/2b74b37b-ac42-4b75-90d3-51c5bf10bcfb">

**Comment by hffvld** [comment]
Verified on `Pixel 7` using version(s):
```
Device/OS: Pixel 7 / panther_beta-user 14 AP11.231215.009 release-keys
Brave build: 1.63.146
Chromium: 121.0.6167.139 (Official Build) beta (64-bit) 
```
Filed follow-up issue #
---------------------------

**STEPS:**
1. Install and Launch Brave `1.63.146`
2. Go to `brave://adblock` OR Three-dot menu > Settings > Brave Shields & privacy > Content Filtering > Create custom filters
3. Add `||example.com^$removeparam=test` to the `Custom Filters` section > Save
4. Go to `https://example.com/?test=true` > Verify that `?test=true` must be removed from the URL search bar

**ACTUAL RESULTS:**
- Verify that the URL bar must not contain `?test=true` when `Block trackers & ads` in `Standard` mode
- Observed that `?test=true` is not removed from the URL search bar

---------------------------

|1|2|3|4|
|:-:|:-:|:-:|:-:|
| ![1](https://github.com/brave/brave-browser/assets/128532965/fab65a14-1cb6-4b54-a693-db3bc5bc6f85) | ![2](https://github.com/brave/brave-browser/assets/128532965/fa58dd7a-f0b6-431f-b86d-672caab70cf2) | ![3](https://github.com/brave/brave-browser/assets/128532965/c2a6eef5-6cb2-46f6-9398-063fe0c112b8) | ![4](https://github.com/brave/brave-browser/assets/128532965/e8be1bf6-a5ac-447d-99d2-6fd0d605903e) |
